pkg/kubelet/config/http.go

```diff
@@ -52,6 +52,8 @@ func NewSourceURL(url string, period time.Duration, updates chan<- interface{})
 func (s *sourceURL) run() {
 	if err := s.extractFromURL(); err != nil {
 		glog.Errorf("Failed to read URL: %v", err)
+		// Emit an update with an empty PodList to allow HTTPSource to be marked as seen
+		s.updates <- kubelet.PodUpdate{[]api.BoundPod{}, kubelet.SET, kubelet.HTTPSource}
```

Comment:
Same idea here-- should only emit this blank list when you get a blank return; a 404 or transport error could maybe be transient and you don't want to mistake that for positive knowledge that nothing should be running.